numpy/f2py/tests/test_crackfortran.py

```diff
@@ -33,3 +36,50 @@ def test_module(self):
         self.module.subc([w, k])
         assert_array_equal(k, w + 1)
         assert self.module.t0(23) == b'2'
+
+class TestPublicPrivate():
+    def test_defaultPrivate(self, tmp_path):
+        f_path = tmp_path / "mod.f90"
+        with f_path.open('w') as ff:
+            ff.write(textwrap.dedent("""\
+            module foo
+              private
+              integer :: a
+              public :: setA
```

Comment:
If this were:
```suggestion
            module foo
              private
              integer :: a
              public :: setA
              integer :: b
```
should `b` be public or private? Is this worth adding a test for?

Reply:
In that case, b would be private, because there is no access specifier given on the definition line of b (it's just specified it is an integer), so it will have the default access of the module, which is private here.

In Fortran, the access specifiers apply on all elements of the module (if not overridden per element), not only the ones after. So if it works for a, it will work for b.

That being said, it's easy to add a test if you believe it's necessary.

Reply:
yes, please

Reply:
That's it, suggestion added in the test.